### PR TITLE
Use forge fmt & lint-staged

### PR DESCRIPTION
**Pull request by Rubilmax**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,14 +2,56 @@
...
+    "commands": [
+      {
+        "match": ".sol",
+        "isAsync": true,
+        "cmd": "forge fmt ${file}"
+      }
+    ]
+  }
+}
+```
```
Comment (MathisGD) [inline README.md:51]: Not mandatory to have this here, but ok

File: package.json [hunk 2]
```diff
@@ -7,10 +7,15 @@
   "author": "MerlinEgalite <merlin@morpho.xyz>",
   "license": "AGPL-3.0",
   "scripts": {
-    "lint": "prettier --write **.sol"
+    "postinstall": "husky install"
```
Comment (MathisGD) [inline package.json:10]: what is this useful for ?

**Review by MathisGD — COMMENTED** [review]
I think that `forge fmt` works well, but if we don't use it everywhere it might be annoying no ?

File: README.md (hunk 1 above)
Comment (MerlinEgalite) [reply README.md:51]: Where do you want to have it?

File: README.md [hunk 3]
```diff
@@ -2,14 +2,56 @@
 
 Repository gathering useful libraries and contracts.
 
-## Setup
+## Getting started
+
+Install this repository as a forge dependency:
+
+```bash
+forge install https://github.com/morpho-dao/morpho-utils
```
Comment (MerlinEgalite) [inline README.md:10]:
```suggestion
forge install morpho-dao/morpho-utils
```

It's a bit shorter

**Review by MerlinEgalite — APPROVED** [review]
I'm with the changes even though it's a bit annoying that we cannot do the following
```solidity
if (true) return 100;
else return 0;
```

File: README.md (hunk 1 above)
Comment (MathisGD) [reply README.md:51]: idk, nowhere ^^
people do whatever they want with their setups

**Review by pakim249CAL — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (MerlinEgalite) [reply README.md:51]: Ok I think it's more like a tip than something mandatory so I don't it's a pb

Comment (MathisGD) [reply README.md:51]: yes

File: package.json (hunk 2 above)
Comment (Rubilmax) [reply package.json:10]: To install git hooks!

**Comment by Rubilmax** [comment]
> I think that `forge fmt` works well, but if we don't use it everywhere it might be annoying no ?

Why would it be annoying?

**Comment by MathisGD** [comment]
>Why would it be annoying?

Because you need to switch the configuration of your IDE each time you change of repo

**Comment by Rubilmax** [comment]
> Because you need to switch the configuration of your IDE each time you change of repo

Fortunately, VSCode & IntelliJ both provide such features
And open-source contributors (outside the Morpho Labs team) are used to such changes in setup (because all open-source repositories don't use the same formatter/formatting settings)

So I'd not consider this

**Comment by MathisGD** [comment]
You mean that you can set a different formatter for each repo on VSCode ? Without the annoying setting file ?

**Comment by Rubilmax** [comment]
> You mean that you can set a different formatter for each repo on VSCode ? Without the annoying setting file ?

Lol no, via the settings file

**Comment by MathisGD** [comment]
Ok, It's manageable though

Then as you want, but I think that it is easier if we change on all repos

**Review by MathisGD — APPROVED** [review]

**Pull request merged**